word_guess.rb

```diff
@@ -0,0 +1,325 @@
+require "colorize"
+class GamePlay
+  attr_reader :number_of_guesses #our output needs to read this for the loop
+
+  def initialize
+    @number_of_guesses = 10
+    @answer_word = WordFetcher.new
+    @answer_validation_array = @answer_word.get_word
+    @user_guess = GuessPrompter.new
+    @guessed_things = []
+  end
+
+  def play_game
+    if win?(@guessed_things)
+      exit
+    end
+
+    if lose?
+      exit
+    end
+
+    print_answer_blanks
+    display_guessed_letters
+    ask_user_for_guess
+
+    if guessed_word?
+      if win?(@last_guess_array)
+        exit
+      end
+    end
+
+      add_to_guess_array
+      update_remaining_guesses
+
+  end
+
+  def win?(array) #my collarborator was a match teacher and she undertands intersections!
+    if @answer_validation_array & array ==  @answer_validation_array.uniq # must use uniq beacause intersection won't return duplicate letters. Otherwise can never win the game!(with words that have duplicate letters)
+      puts "You managed to keep the dinosaur at bay and correctly guessed the word, '#{ @answer_validation_array.join("") }'." #not-match teacherese: when the guessed letter array includes all the letters in the answer array, the "intersection" will be the answer array.
+      puts "You win!"
+      return true
+    else
+      return false
+    end
+  end
+
+  def lose?
+    if @number_of_guesses == 0
+      puts "The dinosaur chomps you before you are able to guess the word, '#{ @answer_validation_array.join("") }'."
+      puts "You were eaten :("
+      return true
+    else
+      return false
+    end
+  end
+
+  def print_answer_blanks
+    puts "==================================================================\n"
+    print "                            " #to move the word blanks away from left justified
+    @answer_validation_array.each do |letter|
+      if @guessed_things.include? letter
```

Comment:
We _can_ use the `include?` method with no parenthesis, but we try to use best practice and include them: `include?(letter)`